src/cursive.rs

```diff
@@ -77,28 +77,38 @@ impl<F: FnOnce(&mut Cursive) -> () + Send> CbFunc for F {
     }
 }
 
-#[cfg(feature = "crossterm-backend")]
+#[cfg(all(
+not(feature = "termion-backend"),
```

Comment:
You don't have to make this one depend on `not(feature = "termion-backend")`. Otherwise, when there is both `crossterm-backend` and `termion-backend`, there will be no `Default` implementation.

In general, the conditions pattern I'm trying to get is this (in pseudo-code):

```
if A {
    Default = A
} else if B {
    Default = B
} else if C {
    Default = C
} else {
    ...
}
```

Only there is no `else` clause for `#cfg` blocks, so we have to unroll it manually, and have:

```
if A {
    Default = A
}

if B and not A:
    Default = B
}

if C and not B and not A {
    Default = C
}

...
```

Reply:
You can use the [`cfg-if`](https://docs.rs/cfg-if/0.1.7/cfg_if/) crate for that:

```rust
if_cfg! {
    if #[cfg(feature = "crossterm-backend")] {
        // Default = crossterm
    } else if #[cfg(feature = "termion-backend")] {
        // Default = termion
    } else if #[cfg(feature = "pancurses-backend")] {
        // Default = pancurses
    } else if #[cfg(feature = "blt-backend")] {
        // Default = blt
    } else if #[cfg(feature = "ncurses-backend")] {
        // Default = ncurses
    }
}
```

By the way, what should be done in case if no features have been enabled? Cursive itself seems to support that case, since it builds with no default features, but `impl Default for Cursive` is missing in that case.

Reply:
Ah right, we can always defaults to the `DummyBackend` if no other built-in backends are available, though failing to compile is not such a bad option, as it often indicates a mistake somewhere (using the dummy backend through `Cursive::default()` is rarely what the user wants).

Reply:
`cfg-if` looks neat, it's just a shame it breaks `rustfmt` inside the macro block :(